### DISCUSSION TITLE
Support of encrypted token when Keycloak used as Idp

### DISCUSSION THREAD

**Discussion by ahcamy2021** [keycloak/keycloak#17355] [Ideas]
Hi,

We noticed that in the latest versions of Keycloak we have some new features that support encpted tokens: encrypted id token, encrypted userinfo token, jwks with public keys used for encryption. But, on the other hand, we noticed that using keycloak as broker IDP (IDentity Provider) is not supporting  encrypted token, do you think it should be added to your backlog as a new feature?

Thanks in advance,

**Comment by darius-m** [comment]
When mentioning tokens and JWKs, do you refer to using OIDC as the protocol, while IdP is referring to SAML? If so, Keycloak does have the option to encrypt assertions, if the client has an encryption key configured.
